pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authentication/AuthenticationProviderToken.java

```diff
@@ -146,10 +164,18 @@ public void initialize(ServiceConfiguration config) throws IOException, IllegalA
 
         long allowedSkew = getConfTokenAllowedClockSkewSeconds(config);
 
-        this.parser = Jwts.parserBuilder()
-                .setAllowedClockSkewSeconds(allowedSkew)
-                .setSigningKey(this.validationKey)
-                .build();
+        JwtParserBuilder jwtParserBuilder = Jwts.parserBuilder().setAllowedClockSkewSeconds(allowedSkew);
+        if (validationKey != null) {
+            jwtParserBuilder.setSigningKey(validationKey);
+        } else {
+            String jwksConfig = config.getProperties().getProperty(confTokenKeySetSettingName);
+            if (StringUtils.isEmpty(jwksConfig)) {
+                throw new IOException("No jwks/secret/Public key was provided for token authentication");
```

Comment:
maybe AuthenticationException?

Reply:
In the old version, here throws `IOException`.